fixtures/coverall/tests/bindings/test_coverall.kts

```diff
@@ -28,4 +30,37 @@ fun Double.almostEquals(other: Double) = Math.abs(this - other) < 0.000001
 assert(d.float32.almostEquals(1.2345F))
 assert(d.maybeFloat32!!.almostEquals(22.0F/7.0F))
 assert(d.float64.almostEquals(0.0))
-assert(d.maybeFloat64!!.almostEquals(1.0))
\ No newline at end of file
+assert(d.maybeFloat64!!.almostEquals(1.0))
+
+// This tests that the UniFFI-generated scaffolding doesn't introduce any unexpected locking.
+// We have one thread busy-wait for a some period of time, while a second thread repeatedly
+// increments the counter and then checks if the object is still busy. The second thread should
+// not be blocked on the first, and should reliably observe the first thread being busy.
+// If it does not, that suggests UniFFI is accidentally serializing the two threads on access
+// to the shared counter object.
+
+ThreadsafeCounter().use { counter ->
```

Comment:
I removed some layers of abstraction from this test, and moved it from "examples" to "fixtures". The abstractions seemed less necessary now that it's not trying to test both the threadsafe and non-threadsafe scenarios.